apps/admin_api/test/admin_api/v1/controllers/blockchain_balance_controller_test.exs

```diff
@@ -0,0 +1,272 @@
+
+      assert response["success"] == true
+
+      balances =
+        Enum.map(response["data"]["data"], fn balance ->
+          {balance["token"]["blockchain_address"], balance["amount"]}
+        end)
+
+      assert length(balances) == 2
+      assert Enum.member?(balances, {"0x00", 123})
+      assert Enum.member?(balances, {"0x02", 123})
+    end
+
+    test_with_auths "filters out inexistent tokens when specifying ids" do
+      blockchain_wallet = insert(:blockchain_wallet, %{address: "0x123"})
+
+      _token_1 = insert(:token, %{id: "tkn_1", blockchain_address: "0x00"})
+      _token_2 = insert(:token, %{id: "tkn_2", blockchain_address: "0x01"})
+      _token_3 = insert(:token, %{id: "tkn_3", blockchain_address: "0x02"})
+
+      attrs = %{
+        "address" => blockchain_wallet.address,
+        "token_ids" => ["tkn_1", "tkn_2", "tkn_4"]
+      }
+
+      response = request("/blockchain_wallet.get_balances", attrs)
+
+      assert response["success"] == true
+
+      balances =
+        Enum.map(response["data"]["data"], fn balance ->
+          {balance["token"]["blockchain_address"], balance["amount"]}
+        end)
+
+      assert length(balances) == 2
+      assert Enum.member?(balances, {"0x00", 123})
+      assert Enum.member?(balances, {"0x01", 123})
+    end
+
+    test_with_auths "filters out non-blockchain tokens when specifying ids" do
+      blockchain_wallet = insert(:blockchain_wallet, %{address: "0x123"})
+
+      _token_1 = insert(:token, %{id: "tkn_1", blockchain_address: "0x00"})
+      _token_2 = insert(:token, %{id: "tkn_2", blockchain_address: nil})
+      _token_3 = insert(:token, %{id: "tkn_3", blockchain_address: "0x02"})
+
+      attrs = %{
+        "address" => blockchain_wallet.address,
+        "token_ids" => ["tkn_1", "tkn_2", "tkn_3"]
+      }
+
+      response = request("/blockchain_wallet.get_balances", attrs)
+
+      assert response["success"] == true
+
+      balances =
+        Enum.map(response["data"]["data"], fn balance ->
+          {balance["token"]["blockchain_address"], balance["amount"]}
+        end)
+
+      assert length(balances) == 2
+      assert Enum.member?(balances, {"0x00", 123})
+      assert Enum.member?(balances, {"0x02", 123})
+    end
+
+    test_with_auths "returns an error when given a non-existing address" do
+      response = request("/blockchain_wallet.get_balances", %{"address" => "0x000"})
+
+      refute response["success"]
+      assert response["data"]["code"] == "unauthorized"
+    end
+
+    test_with_auths "returns an error when address is missing" do
+      response = request("/blockchain_wallet.get_balances", %{})
+
+      refute response["success"]
+      assert response["data"]["code"] == "client:invalid_parameter"
+    end
+
+    test_with_auths "returns an error when there was a problem communicating with the adapter" do
+      Application.put_env(:eth_blockchain, :default_adapter, :invalid_adapter)
```

Comment:
This will cause flaky tests unless `use AdminAPI.ConnCase, async: false`. It's not happening now probably because not many tests rely on `:default_adapter` config yet.

Reply:
Wait, I realized this and I thought I removed this already, my bad.